### PR TITLE
Added AzureBlobFileSystem support for StructuredDatasets

### PR DESCRIPTION
**Pull request by MorpheusXAUT**
# TL;DR
This PR adds support for storing `StructuredDatasets` using AzureBlobFileSystem (`abfs`).

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
As discussed before, we've added support for `abfs` (using `adlfs`/`stow` under the hood) for `StructuredDatasets` by adding it to the registered protocols for transformers.
I've also noticed one file (`plugins/flytekit-spark/flytekitplugins/spark/sd_transformers.py`) which still used string constants and didn't support GCS either. As we're currently not using Spark anywhere, I wasn't able to verify this change though, so I can revert those lines if you'd prefer.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2709

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by wild-endeavor** [comment]
hey @MorpheusXAUT do you also have the data persistence plugin for abfs?  is that what you meant by you previously added support for it under the hood?

**Comment by MorpheusXAUT** [comment]
> hey @MorpheusXAUT do you also have the data persistence plugin for abfs? is that what you meant by you previously added support for it under the hood?

@wild-endeavor oh, good point 🤔 I believe that still hasn't been cleaned up & contributed back to this repo, sorry about that.
I'll try to get that sorted ASAP. Should we add that to the same PR or create a new one?

**Comment by wild-endeavor** [comment]
this pr is fine.  thanks!

**Comment by MorpheusXAUT** [comment]
@wild-endeavor Looking at it again, I was actually mistaken last night.

We don't use an extra (custom-written) datapersistence plugin for `abfs`, but instead make use of `fsspec` and its [`adlfs`](https://github.com/fsspec/adlfs) (albeit a [fork](https://github.com/blackshark-ai/adlfs/tree/fix-expand-paths-using-globs) at the moment, since we're waiting for a [PR](https://github.com/fsspec/adlfs/pull/305) to be merged).

I've added `adlfs` to `flytekit-data-fsspec` plugin's `extras_require` map - I believe that's everything required on flytekit's/the data persistance plugin's side as the actual import for `adlfs` was configured in our projects.

**Review by pingsutw — APPROVED** [review]

**Comment by wild-endeavor** [comment]
rerunning the failed onnx job.  not entirely sure what's happening there.

File: flytekit/types/structured/structured_dataset.py [hunk 1]
```diff
@@ -39,6 +39,7 @@
 # Protocols
 BIGQUERY = "bq"
 S3 = "s3"
+ABFS = "abfs"
 GCS = "gs"
 LOCAL = "/"
```
Comment (eapolinario) [inline flytekit/types/structured/structured_dataset.py:44]: Is it time to turn this into an enum, @wild-endeavor ?
